projects/dev/src/app/group/group-content/group-content.component.ts

```diff
@@ -1,39 +1,101 @@
-import { Component, OnInit, ElementRef, EventEmitter, Input, Output } from '@angular/core';
+import { Component, OnInit, ElementRef, Input } from '@angular/core';
 import * as _ from 'lodash';
+import { ActivatedRoute, Router, UrlSegment } from '@angular/router';
+import { Group } from '../../shared/models/group.model';
+import { Location } from '@angular/common';
+
+export enum Management {
+  None = 'none',
+  MembershipsAndGroup = 'memberships_and_group'
+}
+
+export enum TabUrls {
+  Overview = '',
+  Composition = 'members',
+  Administration = 'managers',
+  Settings = 'settings'
+}
 
 @Component({
   selector: 'app-group-content',
   templateUrl: './group-content.component.html',
   styleUrls: ['./group-content.component.scss'],
 })
 export class GroupContentComponent implements OnInit {
-  @Input() data;
-  @Input() empty;
 
-  @Output() expandWholeWidth = new EventEmitter<void>();
+  @Input() group: Group;
+  activeTab = 0;
 
-  constructor(private elementRef: ElementRef) {}
+  constructor(
+    private activatedRoute: ActivatedRoute,
+    private elementRef: ElementRef,
+    private router: Router,
+    private location: Location) {
+  }
 
-  ngOnInit() {}
+  ngOnInit() {
+    // tslint:disable-next-line
+    this.activatedRoute.url.subscribe(_ => {
+      const path = this.location.path().split('/').pop();
+      switch (path) {
+        case TabUrls.Composition:
+          this.activeTab = 1;
+          break;
+        case TabUrls.Administration:
+          this.activeTab = 2;
+          break;
+        case TabUrls.Settings:
+          this.activeTab = 3;
+          break;
+        default:
+          this.activeTab = 0;
+          break;
```

Comment:
What's your suggestion?

Reply:
There is a question "why isn't it ..." in the comment.

Reply:
Then I need to repeat the same structure of the tabs 4 times.
Of course each component will only have one content fillable and the others are empty.

Reply:
Does it work for you?

Reply:
No, in what I describe, everything in common is in the group content (should probably have a better name) component. This component includes group title, pending request component, tab bar and then a GroupContentTabContent component which itself has a router.

Reply:
Then we need to change the structure of current code - previous code from `design` sub app isn't necessary at all for now

Reply:
But isn't it the clean way to do this in Angular?

Reply:
Hmm, in that case we should use `ngrx` store to avoid multiple api endpoint call.
Because routed components can't get parent's data.
In this case, the routed components from group content component can't get the `group` info so we should keep that in the `ngrx` store for further usage.

Reply:
Ok, I understand. NgRx is quite complex for that purpose and quite "big" without knowing exactly what will be our future needs (and would require good testing).
Isn't it what they are doing here: https://angular.io/guide/router#fetch-data-before-navigating ?

Reply:
Then we need to save the group in Subject to prevent call to the api endpoint.
Anyway calling before navigation also calls the `getGroup` service which means duplication of api endpoint call.

Reply:
> Then we need to save the group in Subject variable to prevent call to the api endpoint.

Looks much easier than using NgRx

>  Anyway calling before navigation also calls the getGroup service which means duplication of api endpoint call.

I don't get what you mean by that. Before what navigation?

Reply:
I am talking about the link you sent me, it fetches the data and provides it to the routed components before navigation which calls the api endpoint using service and it causes duplication of api endpoint call.

Reply:
So, isn't it a more appropriate way to do (use a router + this rx subject for group info) ? 
Components should not deal with routing.